pkg/log/log.go

```diff
@@ -0,0 +1,100 @@
+// Package log provides a logger.
+package log
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Logger is the interface used for all loggers in Juno.
+type Logger interface {
+	Debug(...any)
+	Debugw(string, ...any)
+
+	Info(...any)
+	Infow(string, ...any)
+
+	Error(...any)
+	Errorw(string, ...any)
+
+	Named(string) Logger
+}
+
+// Log wraps a *zap.SugaredLogger to fulfill the Logger interface.
+type Log struct {
+	zapLogger *zap.SugaredLogger
+}
+
+// *Log implements Logger
+var _ Logger = &Log{}
+
+func NewNopLogger() *Log {
+	return NewLogger(zap.NewNop().Sugar())
+}
+
+// NewProductionLogger creates a *Log with sane defaults.
+func NewProductionLogger(verbosity string) (*Log, error) {
+	re := regexp.MustCompile("(?i)debug|info|error")
+	if !re.Match([]byte(verbosity)) {
+		return nil, fmt.Errorf("cannot parse verbosity: %s", verbosity)
+	}
```

Comment:
I don't see a place where it would be superior to simply passing an error up the stack. If you see a place where Fatal would be the best solution, I can add it.